pkg/queue/bucketing_test.go

```diff
@@ -252,6 +253,87 @@ func TestDescendingRecord(t *testing.T) {
 	}
 }
 
+func TestRequestsSuddenStop(t *testing.T) {
+	RegisterTestingT(t)
+	now := time.Date(2024, 6, 26, 12, 0, 0, 0, time.UTC)
+	buckets := NewRequestsBuckets(5*time.Second, granularity)
+
+	// empty window
+	Expect(buckets.buckets).To(Equal([]int{0, 0, 0, 0, 0}))
+	Expect(buckets.WindowAverage(now)).To(Equal(0.0))
+
+	// first bucket with 1 request
+	buckets.Record(now, 1)
+	Expect(buckets.WindowAverage(now)).To(Equal(1.0))
+	Expect(buckets.buckets).To(Equal([]int{1, 0, 0, 0, 0}))
+
+	// second bucket with 2 requests
+	buckets.Record(now.Add(1*time.Second), 2)
+	Expect(buckets.WindowAverage(now.Add(1 * time.Second))).To(Equal(1.5))
+	Expect(buckets.buckets).To(Equal([]int{1, 2, 0, 0, 0}))
+
+	// third bucket with 3 requests
+	buckets.Record(now.Add(2*time.Second), 3)
+	Expect(buckets.WindowAverage(now.Add(2 * time.Second))).To(Equal(2.0))
+	Expect(buckets.buckets).To(Equal([]int{1, 2, 3, 0, 0}))
+
+	// fourth bucket with 4 requests
+	buckets.Record(now.Add(3*time.Second), 4)
+	Expect(buckets.WindowAverage(now.Add(3 * time.Second))).To(Equal(2.5))
+	Expect(buckets.buckets).To(Equal([]int{1, 2, 3, 4, 0}))
+
+	// fifth bucket with 5 requests
+	buckets.Record(now.Add(4*time.Second), 5)
+	Expect(buckets.WindowAverage(now.Add(4 * time.Second))).To(Equal(3.0))
+	Expect(buckets.buckets).To(Equal([]int{1, 2, 3, 4, 5}))
+
+	// first bucket (sixth time window), we don't have any requests, so the average should be 0+2+3+4+5/5 = 2.8
+	// but the buckets don't change until new value is recorded or until the window expires
+	Expect(buckets.WindowAverage(now.Add(5 * time.Second))).To(Equal(2.8))
```

Comment:
without this change, this would have been `(2+3+4+5)/(5-1) = 3.5` which means increase from `3.0` after just receiving 0 requests in the sixth time window